common/cuda_hip/factorization/lu_kernels.cpp

```diff
@@ -85,7 +85,7 @@ __global__ __launch_bounds__(default_block_size) void initialize(
 }
 
 
-template <typename ValueType, typename IndexType>
+template <bool checked_lookup, typename ValueType, typename IndexType>
 __global__ __launch_bounds__(default_block_size) void factorize(
```

Comment:
I think I would prefer a name that states what it is meant to be used for, rather than a name describing an implementation detail. Something like `has_all_fillin`?

Reply:
I also prefer not safe_lookup. I will think why I use unsafe version rather than safe version.